src/v2/Apps/Order/Components/OfferHistoryItem.tsx

```diff
@@ -32,7 +33,7 @@ const OfferHistoryItem: React.SFC<
       <Row>
         <div />
         <Sans size="2" color="black60">
-          List price: {totalListPrice}
+          List price: {(offerItem as any).price}
```

Comment:
This is the casting I'm trying to avoid but not sure how. Without it I get the following warning:
<img width="421" alt="Screen Shot 2020-11-23 at 3 54 53 PM" src="https://user-images.githubusercontent.com/437156/100021339-585dc880-2daf-11eb-8aa4-3853190c98b1.png">

Reply:
I would recommend wrapping this whole `Sans` in a logic statement. 

```jsx
{__typename !== "%other" && <Sans ... />}
```

Typescript would narrow down the type definition in this case to be `Artwork`, `EditionSet` or potentially any newly defined type! If the type wasn't defined explicitly on the union though (it had a type of `%other`) then it wouldn't try to render this field which would avoid a potential error. If someone came along and added an explicit type that _didn't_ have the field value, then typescript would throw and error b/c while `%other` would be narrowed out in the types, the all other types would still be checked against the access of price. At that time the logic could be made more robust if necessary.

Reply:
@zephraph, @sweir27 - so I tried to just plainly check for what you've suggested `{__typename !== "%other" && <Sans ... />}` and it did not clear the type script warning about the potential that the properties would not exist on 'other' type.

So I've tried to add this helper function that narrows the `OfferItem` type. Not sure if that makes sense here. Please let me know what you guys think of that implementation?

Reply:
I was able to get the warning to go away with a line like:
```typescript
const item = order.lineItems.edges[0].node.artworkOrEditionSet.__typename !== "%other" && order.lineItems.edges[0].node.artworkOrEditionSet
const itemPrice = item.price
```

But this isn't so different from the helper method you made!

Reply:
Helper method also helps to not repeat this line in 4 places:) Also I think I prefer the check to explicit artwork of edition set type as a more clear statement that we only support those 2 at that point in time.